convert2rhel/systeminfo.py

```diff
@@ -320,38 +325,62 @@ def get_enabled_rhel_repos(self):
 
-        This method will try to estabilish a socket connection through the
-        default host in the method signature (8.8.8.8). If it's connected
-        successfully, then we know that internet is accessible from the host.
+        This method will try to retrieve a web page on the Red Hat network that
+        we know to exist (http://static.redhat.com/test/rhel-networkmanager.txt).
+        If we can successfully retrieve that page, validate that the status_code
+        is 200, and assert that the content of the response is the string 'OK',
+        then we decide we are connected to the internet.
+
+        We check a web page because we will need working https to retrieve
+        packages from Red Hat infrastructure during the conversion.
 
         .. warnings::
             We might have some problems with this if the host machine is using
-            a NAT gateway to route the outbound requests
-
-        .. seealso::
-            * Comparison of different methods to check internet connections: https://stackoverflow.com/a/33117579
-            * Redirecting IP addresses: https://superuser.com/questions/954665/how-to-redirect-route-ip-address-to-another-ip-address
-
-        :param host: The host to establish a connection to. Defaults to "8.8.8.8".
-        :type host: str
-        :param port: The port for the connection. Defaults to 53.
-        :type port: int
-        :param timeout: The timeout in seconds for the connection. Defaults to 3.
-        :type port: int
-        :return: Return boolean indicating whether or not we have internet access.
+            a NAT gateway to route the outbound requests to any other service.
+
+            DNS could also be used to redirect the URL we test to another address.
+
+        :return: Return boolean indicating whether or not we have internet
+            access.
         :rtype: bool
         """
-        try:
-            self.logger.info("Checking internet connectivity using host '%s' and port '%s'." % (host, port))
-            socket.setdefaulttimeout(timeout)
-            socket.socket(socket.AF_INET, socket.SOCK_STREAM).connect((host, port))
-            return True
-        except (socket.timeout, socket.error):
-            self.logger.warning("Couldn't connect to the host '%s'." % host)
-            return False
+        retries = 0
+        is_reachable = False
+        self.logger.info(
+            "Checking internet connectivity using address '%s'.",
+            CHECK_INTERNET_CONNECTION_ADDRESS,
+        )
+        while retries <= CHECK_INTERNET_CONNECTION_RETRIES:
+            try:
+                response = urllib.request.urlopen(CHECK_INTERNET_CONNECTION_ADDRESS)
+            except urllib.error.URLError as err:
+                self.logger.debug("Faild to retrieve data from host, reason: %s", err.reason)
+                break
+
+            status_code = response.getcode()
+            text_response = response.read().decode("utf-8")
```

Comment:
@abadger, do you think that this part can fail? I mean, if the connection is successful and the host doesn't return anything to us (probably they have deployed a new version of that server, and it's misbehaving), do you think that we need to extra careful here and do another try-except check? 

I mean, what are the chances of something like that happening? I can see the scenario where this host is offline for some reason, but not having its content changed.